inference_rules.adoc

```diff
@@ -757,54 +747,103 @@ The auditor may also request source code access to binary elements of the submis
+
+F(h; n, p) = ∑ f(k; n, p),
+ 
+with the summation going from k = h to n.
+
+Note that, holding h and n fixed, this probability increases with p. This is because, as p gets larger, the odds that our n queries produced results at least as good as h successes and t failures increases. In other words, it is easier to achieve a larger number of successes when the underlying probability of an individual success is higher. 
+
+The this cumulative distribution function for the binomial distribution, F(k; n, p), can be written in terms of the regularized incomplete beta function. The (unregularized) incomplete beta function is defined as:
+
+B(x; a, b) = ∫t^(a - 1) * (1-t)^(b-1) dt,
+where the integral goes from 0 to x.  
+
+We can regularize this to attain:
+
+I(x; a, b) = B(x; a, b) / B(1; a, b).
+
+Note that this is "regularized" in the sense that I(0; a, b) = 0, and I(1; a, b) = 1. 
+
+We have an alternate expression for F(k; n, p) in terms of this function:
+
+F(k; n, p) = I(1 - p; n - k, k + 1).
+
+Since the regularized incomplete beta function can be estimated via a continued fraction or by evaluating the Gaussian hypergeometric function, this provides a method for efficiently computing the cumulative density function, F(k; n, p).
+
+=== Determining the Early Stopping Criterion
+
+We can use the computation from the previous section to derive an early stopping condition for performing queries to determine whether a system meets a latency bound. Suppose a benchmark requires that our system meets the latency bound p percent of the time. Given that we have seen t queries which are overlatency, at least how many underlatency queries must we see to be sure—within a certain confidence threshold—that we achieve the desired latency bound?
+
+Fix the following variables:
+
+* p = the percentile for our tail latency (the percentage of the time we would like our system to achieve the set latency bound)
+* c = confidence (1 - (false positive rate for minimally failing system))
+* d = tolerance (amount below target success rate for minimally failing system)
+* t = number of overlatency queries seen thus far.
+
+We need to determine the smallest h (number of underlatency queries) so that the likelihood of seeing at most t overlatency queries less than 1-c. This is given by an expression involving the cumulative distribution function from the previous section:
+
+F(t; h + t, 1 - (p - d)) <= 1-c.
+
+The left hand side is the probability that the minimally failing system (i.e. one with underlatency rate p-d) resulted in t or fewer overlatency queries. Intuitively, we want to know the smallest number of underlatency queries required such that the probability of us seeing this good of a result, assuming a minimally failing system, is very low (at most 1-c). In other words, in order for us to have seen such a good result, we should be quite sure that we meet the latency bound.
```

Comment:
Should "t or fewer overlatency queries" be "t or more latency queries"?

Reply:
I think it should still be "t or fewer overlatency queries".

The lefthand side of the equation should be the odds that a minimally-failing system performs at least as well as having t overlatency queries, which happens when it has at most t overlatency.

Reply:
@DilipSequeira @ckstanton Can you resolve the question?

Reply:
I went back and looked at the rest of the explanation.  I believe it should still be "t or fewer overlatency queries" here, but the definition of the cumulative distribution function from earlier is wrong (in particular, F(h; n, p) should be the probability of at MOST h successes, not at LEAST).  That should fix the confusion with this line.  I'll go back and fix that.  Thanks for the catch, @DilipSequeira !

Reply:
The definition of the CDF has been fixed (so it now agrees with "t or fewer overlatency queries" in this line).  This PR should be ready to go, if there are no other objections.